crates/lib3h/src/engine/ghost_engine.rs

```diff
@@ -402,109 +402,107 @@ impl<'engine> GhostEngine<'engine> {
+                                let can = !includes(&known_aspects, &aspect_address_list);
+                                can
+                            }
+                        };
+                        if can_fetch {
+                            let msg_data = FetchEntryData {
+                                space_address: space_address.clone(),
+                                entry_address: entry_address.clone(),
+                                request_id: me.request_track.reserve(),
+                                provider_agent_id: provider_agent_id.clone(),
+                                aspect_address_list: None,
+                            };
+
+                            /*                              let _context = RequestContext {
+                                space_address: msg.space_address.to_owned(),
+                                agent_id: msg.provider_agent_id.to_owned(),
+                            };*/
+                            let _ = me.lib3h_endpoint.request(
+                                Lib3hSpan::todo(),
+                                Lib3hToClient::HandleFetchEntry(msg_data),
+                                Box::new(move |me, response| {
+                                    let space_gateway = me
+                                        .get_space(
+                                            &space_address.to_owned(),
+                                            &provider_agent_id.to_owned(),
+                                        )
+                                        .map_err(|e| GhostError::from(e.to_string()))?;
+                                    match response {
+                                        GhostCallbackData::Response(Ok(
+                                            Lib3hToClientResponse::HandleFetchEntryResult(msg),
+                                        )) => {
+                                            space_gateway.publish(
+                                                Lib3hSpan::todo(),
+                                                GatewayRequestToChild::Dht(
+                                                    DhtRequestToChild::BroadcastEntry(
+                                                        msg.entry.clone(),
+                                                    ),
+                                                ),
+                                            )?;
+                                        }
+                                        GhostCallbackData::Response(Err(e)) => {
+                                            error!("Got error on HandleFetchEntryResult: {:?} ", e);
+                                        }
+                                        GhostCallbackData::Timeout => {
+                                            error!("Got timeout on HandleFetchEntryResult");
+                                        }
+                                        _ => panic!("bad response type"),
```

Comment:
Send all errors / panics out in the Result, then on lines 577 and 606, send them out in the response